larsim/PhotonPropagation/SemiAnalyticalModel.cxx

```diff
@@ -239,12 +231,12 @@ SemiAnalyticalModel::VUVVisibility(geo::Point_t const& ScintPoint, OpticalDetect
   }
   else {
     throw cet::exception("SemiAnalyticalModel")
-        << "Error: Invalid optical detector shape requested - configuration error in semi-analytical model, only rectangular, dome or disk optical detectors are supported." << "\n";
+      << "Error: Invalid optical detector shape requested - configuration error in semi-analytical model, only rectangular, dome or disk optical detectors are supported." << "\n";
   }
 
   // calculate visibility by geometric acceptance
   // accounting for solid angle and LAr absorbtion length
-  double visibility_geo = std::exp(-1. * distance / fL_abs_vuv) * (solid_angle / (4 * CLHEP::pi));
+  double visibility_geo = std::exp(-1. * distance / fvuv_absorption_lenght) * (solid_angle / (4 * CLHEP::pi));
```

Comment:
We understand that you did not change the logic here (just the name of the variable).  But is there any possibility of dividing by zero?  Or does the interpolation done above ensure that that can't happen?

Reply:
There is some possibility, depending on the input parameters. I've encapsulated the calculation on a function, check that is not 0 or less and made the variable a `const`.